synapse/tests/test_lib_base.py

```diff
@@ -331,3 +331,28 @@ def test_base_main_sigint(self):
 
         proc.join(timeout=10)
         self.eq(proc.exitcode, 137)
+
+    async def test_onwith(self):
```

Comment:
haha, just couldn't resist it :D  I was wondering if i'd see this in it's own PR or slipped into another ;)